src/mono/wasm/debugger/DebuggerTestSuite/Inspector.cs

```diff
@@ -81,12 +81,20 @@ public Task<JObject> WaitFor(string what)
             {
                 if (tcs.Task.IsCompleted)
                 {
+                    Client.CurrentSessionId = new SessionId(tcs.Task.Result["sessionId"]?.Value<string>());
                     notifications.Remove(what, out _);
                     return tcs.Task;
                 }
 
                 throw new Exception($"Invalid internal state, waiting for {what} while another wait is already setup");
             }
+            else if (nextNotifications.TryDequeue(out var notification))
```

Comment:
What if the next notification in the queue is for a different `string what`?

Reply:
I'll push some changes and it will be throwing an exception if it's a different `string what`